pkg/controller/privatelink/actuator/awsactuator/awshubactuator_test.go

```diff
@@ -0,0 +1,1494 @@
+			Name: aws.String(testAPIDomain),
+			ResourceRecords: []*route53.ResourceRecord{{
+				Value: aws.String("0.0.0.1"),
+			}, {
+				Value: aws.String("0.0.0.2")},
+			},
+			TTL:  aws.Int64(10),
+			Type: aws.String("A"),
+		},
+	}, { // There should be an error when configured to use alias target but dnsRecord is nil
+		name:        "Alias, dsnRecord is nil",
+		cd:          cdBuilder.Options(testcd.WithAWSPlatform(&hivev1aws.Platform{})).Build(),
+		config:      &hivev1.AWSPrivateLinkConfig{DNSRecordType: hivev1.AliasAWSPrivateLinkDNSRecordType},
+		expectError: "configured to use alias target, but no alias target found.",
+	}, { // There should be an error when configured to use alias target but AliasTarget.Name is empty
+		name:        "Alias, target name is empty",
+		cd:          cdBuilder.Options(testcd.WithAWSPlatform(&hivev1aws.Platform{})).Build(),
+		config:      &hivev1.AWSPrivateLinkConfig{DNSRecordType: hivev1.AliasAWSPrivateLinkDNSRecordType},
+		record:      &actuator.DnsRecord{},
+		expectError: "configured to use alias target, but no alias target found.",
+	}, { // There should be an error when configured to use alias target but AliasTarget.HostedZoneID is empty
+		name:   "Alias, target hostedZoneID is empty",
+		cd:     cdBuilder.Options(testcd.WithAWSPlatform(&hivev1aws.Platform{})).Build(),
+		config: &hivev1.AWSPrivateLinkConfig{DNSRecordType: hivev1.AliasAWSPrivateLinkDNSRecordType},
+		record: &actuator.DnsRecord{AliasTarget: actuator.AliasTarget{Name: "test-aliastarget"}},
+		//record:        &actuator.DnsRecord{AliasTarget: actuator.AliasTarget{Name: "test-aliastarget", HostedZoneID: "test-hostedzoneid"}},
+		expectError: "configured to use alias target, but no alias target found.",
+	}, { // When configured, a dns Record with AliasTarget should be returned
+		name:   "Alias, result returned",
+		cd:     cdBuilder.Options(testcd.WithAWSPlatform(&hivev1aws.Platform{})).Build(),
+		config: &hivev1.AWSPrivateLinkConfig{DNSRecordType: hivev1.AliasAWSPrivateLinkDNSRecordType},
+		record: &actuator.DnsRecord{AliasTarget: actuator.AliasTarget{Name: "test-aliastarget", HostedZoneID: "test-hostedzoneid"}},
+		expectResult: &route53.ResourceRecordSet{
+			AliasTarget: &route53.AliasTarget{
+				DNSName:              aws.String("test-aliastarget"),
+				EvaluateTargetHealth: aws.Bool(false),
+				HostedZoneId:         aws.String("test-hostedzoneid"),
+			},
+			Name: aws.String(testAPIDomain),
+			Type: aws.String("A"),
+		},
+	}}
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			awsHubActuator := &AWSHubActuator{config: test.config}
+
+			result, err := awsHubActuator.recordSet(test.cd, testAPIDomain, test.record)
+			if test.expectError == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, err, test.expectError)
+			}
+
+			assert.Equal(t, test.expectResult, result)
+		})
+	}
+}
+
+func Test_reconcileHostedZoneAssociations(t *testing.T) {
```

Comment:
I was thinking this would be really difficult to test due to how the newAWSClient works. However, I'm realizing I can still test everything except that.
[Done](https://github.com/openshift/hive/compare/d6f2766aef12b836a75c70e06ffdc2d99246f12e..865e42a710b873165a6fcd6045a50292ad1b1e7c)